uast/uast.go

```diff
@@ -31,316 +31,280 @@ const (
-	// subpackages.
-	// Implies VisibleFromInstance, VisibleFromType and VisibleFromPackage.
-	VisibleFromSubpackage
-	// VisibleFromModule marks modifiers that declare visibility from the
-	// same module (e.g. Java JAR).
-	// Implies VisibleFromInstance and VisibleFromType.
-	VisibleFromModule
-	// VisibleFromFriend marks modifiers that declare visibility from friends
-	// (e.g. C++ friends).
-	// Implies VisibleFromInstance and VisibleFromType.
-	VisibleFromFriend
-	// VisibleFromWorld implies full public visibility. Implies all other
-	// visibility levels.
-	VisibleFromWorld
+	// Add is an arithmetic operator (i.e. `+`).
+	Add
+
+	// Substract in an arithmetic operator (i.e. `-`).
+	Substract
+
+	// Multiply is an arithmetic operator (i.e. `*`).
+	Multiply
+
+	// Divide is an arithmetic operator (i.e. `/`).
+	Divide
+
+	// Modulo is an arithmetic operator (i.e. `%`, `mod`, etc.).
+	Modulo
+
+	// Package indicates that a package level property
+	Package
+
+	// Declaration is a construct to specify properties of an identifier
+	Declaration
+
+	// Import indicates an import level property
+	Import
+
+	// Path is a qualified name of some construct
+	Path
+
+	// Alias is an alternative name for some construct
+	Alias
+
+	// Function is a sequence of instructions packaged as a unit
+	Function
+
+	// Body is a sequence of instructions in a block
+	Body
+
+	// Name is an identifier used to reference a value.
+	Name
+
+	// Receiver is the target of a construct (message, function, etc.)
+	Receiver
+
+	// Argument is variable used as input/output in a function
+	Argument
+
+	// Value is an expression that cannot be evaluated any further.
+	Value
+
+	// ArgsList is variable number of arguments (i.e. `...`, `Object...`, `*args`, etc.).
+	ArgsList
+
+	// Base is the parent type of which another type inherits.
+	Base
+
+	// Implements is the type (usually an interface) that another type implements.
+	Implements
+
+	// Instance is a concrete occurrence of an object.
+	Instance
+
+	// Subtype is a type that can be used to substitute another type
+	Subtype
+
+	// Subpackage is a package that is below another package in the hierarchy
+	Subpackage
+
+	// Module is a set of funcitonality grouped
+	Module
+
+	// Friend is an access granter for some private resources
+	Friend
+
+	// World is a set of every component
+	World
 
 	// If is used for if-then[-else] statements or expressions.
 	// An if-then tree will look like:
 	//
-	// 	IfStatement {
+	// 	If, Statement {
 	//		**[non-If nodes] {
-	//			IfCondition {
+	//			If, Condition {
 	//				[...]
 	//                      }
 	//		}
 	//		**[non-If* nodes] {
-	//			IfBody {
+	//			If, Then {
 	//				[...]
 	//			}
 	//		}
 	//		**[non-If* nodes] {
-	//			IfElse {
+	//			If, Else {
 	//				[...]
 	//			}
 	//		}
 	//	}
 	//
-	// The IfElse node is optional. The order of IfCondition, IfBody and
-	// IfElse is not defined.
+	// The Else node is optional. The order of Condition, Then and
+	// Else is not defined.
```

Comment:
Unrelated: these kind of comments with easy examples of applicability are great. We probably should improve on this on the doc page for creating drivers.

Reply:
Right, it's difficult to do so without loss of generality though.